test/graphql.paymentMethods.test.js

```diff
@@ -15,7 +15,11 @@ import { randEmail } from './stores';
 let host, admin, user, collective, paypalPaymentMethod;
 
 describe('graphql.paymentMethods.test.js', () => {
-  beforeEach(() => utils.resetTestDB());
+  beforeEach(async () => {
+    await utils.resetTestDB();
+    await utils.resetTestModelDb('CollectiveHistory');
```

Comment:
Why do we need to to that? What does that solve?

Reply:
After many tests I came to the conclusion that it can be changed. Its original goal was to ensure the creation of the table "CollectiveHistory" which failed repeatedly.
I think the problem is due to an error in sequelize: sequelize.sync not completed when resolving (https://github.com/sequelize/sequelize/issues/8918) or by asynchronous events from which no resolution is expected (I saw some in the hooks) and that events/functions use the database that is being recreated by sequelize.sync at the same time.

If I add time before and after running sequelize.sync all problems disappear. The amount of time varies depending on the execution environment (memory, processor, Internet connection, etc.).

During the debugging process the only thing I noticed that improved the success rate was to enable the __logQueryParameters__ and __logging__ options. 😯 I suppose it slows down some processes and thereby giving others time to conclude gracefully.

In the case of the file ```graphql.paymentMethods.test``` (where the problem with "CollectiveHistory" occurs) I had to add time before running sequelize.sync, which gives me to understand that some postgres or sequelize process is announced as finished when it actually continues to block writing, or that some hook continues to work asynchronously while table deletion occurs.
_________________
Rather than worrying about the appearance of logs, what I would like to understand is the impact of these lags on production. I don't think there is any creation/droppingse of tables in production, so this is not a threat, but it can be remembered as something to revisit.